cmd/openshift-install/create.go

```diff
@@ -105,19 +105,19 @@ var (
 					logrus.Fatal("Bootstrap failed to complete: ", err)
 				}
 
-				logrus.Info("Destroying the bootstrap resources...")
-				err = destroybootstrap.Destroy(rootOpts.dir)
-				if err != nil {
-					logrus.Fatal(err)
-				}
-
```

Comment:
This change is fine for WIP, but should be removed before final version.

Reply:
@abhinavdahiya The workflow was changed to:
1. wait for the installation to be complete, https://github.com/openshift/installer/pull/2608/files#diff-266f03122200ace2d21f2e449b6a5503R108-R120
2. Destroy the etcd-bootstrap member and remove the etcd-bootstrap member from etcd api. https://github.com/openshift/cluster-etcd-operator/blob/3dc39d7e8bcec453418aa77f1b15ad1095b12e51/pkg/operator/bootstrapteardown/teardown.go#L22

The main motivation for this was to:
1. make sure the etcd-bootstrap is only removed when cluster has bootstrapped. We wanted to avoid etcd-bootstrap getting destroyed asynchronously by the installer while cluster-etcd-operator is still reconciling on minority of members.
2. with the introduction of cluster-etcd-operator, we have one more step in destroying etcd-bootstrap, i.e. removing it from the etcd api. We wanted the removal in cluster-etcd-operator perview of cluster-etcd-operator, solely.

Because of the aforementioned reasons, we hinged on observing the clusterversions and when it reports successful install, ceo removes the etcd-bootstrap and related resources from the cluster while installer removes the vm (both actions happening asynchronously).

Having said that, we will keep an active list of issues to research and try to find a way in which we don't have to keep the bootstrap vm any longer than absolutely needed. 

Is that a valid approach?

Reply:
The bootstrap host needs to be destroyed as soon as possible because that hinders with the control-plane because the apiserver etc are configured differently on bootstrap host and control plane. 

Install complete is also user's hook at the cluster is good to use, changing the etcd member at that point is incorrect.

Also bootstrap host is a privilege entity, the shorter it stays up the better.

So no. The order should not be changed.

The etcd-operator is free to block bootstrapping until it has finished the etcd cluster bootstrap on the control plane.. then the installer can be instructed to destroy bootstrap.